### PR TITLE
[INLONG-10535][Agent] Support minute level tasks

### LINKED ISSUE #10535: [Improve][Agent] Support minute level tasks
### Description

Support minute level tasks

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
Fixes #10535 

### Motivation

Support minute level tasks for file collection 

### Modifications

1 Add m cycle types and modify the corresponding regular matching code
2 Add corresponding unit tests for file scanning

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

No doc needed

### CODE REVIEW

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/CycleUnitType.java [hunk 1]
```diff
@@ -21,5 +21,6 @@ public class CycleUnitType {
 
     public static final String DAY = "D";
     public static final String HOUR = "h";
+    public static final String MINUTE = "m";
```
Comment (aloyszhang) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/CycleUnitType.java:24]: DAY and REAL_TIME are in upper case, while HOUR and MINUTE are in lower case. 
Plz keep consistency.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java [hunk 2]
```diff
@@ -170,34 +170,34 @@ private static Calendar getDateTime(Calendar calendar, String cycleUnit, String
...
             calendar.add(Calendar.DAY_OF_YEAR, offsetNumber);
         }
 
-        if ("h".equalsIgnoreCase(offsetUnit)) {
+        if (CycleUnitType.HOUR.equalsIgnoreCase(offsetUnit)) {
             calendar.add(Calendar.HOUR_OF_DAY, offsetNumber);
         }
 
-        if ("m".equals(offsetUnit)) {
+        if (CycleUnitType.MINUTE.equals(offsetUnit)) {
```
Comment (aloyszhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:200]: equalsIgnoreCase ？

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java [hunk 3]
```diff
@@ -207,15 +207,11 @@ private static Calendar getDateTime(Calendar calendar, String cycleUnit, String
...
             timeInterval = DAY_TIMEOUT_INTERVAL;
-        } else if ("M".equals(cycleUnit)) {
+        } else if (CycleUnitType.HOUR.equalsIgnoreCase(cycleUnit)) {
             timeInterval = HOUR_TIMEOUT_INTERVAL;
-        } else if ("D".equalsIgnoreCase(cycleUnit)) {
-            timeInterval = DAY_TIMEOUT_INTERVAL;
-        } else if ("h".equalsIgnoreCase(cycleUnit)) {
-            timeInterval = HOUR_TIMEOUT_INTERVAL;
-        } else if (cycleUnit.contains("m")) {
+        } else if (cycleUnit.contains(CycleUnitType.MINUTE)) {
```
Comment (aloyszhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:214]: why `cycleUnit.contains` here?

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/DateTransUtils.java [hunk 4]
```diff
@@ -45,15 +47,11 @@ public static long timeStrConvertToMillSec(String time, String cycleUnit, TimeZo
             throws ParseException {
         long retTime = 0;
         SimpleDateFormat df = null;
-        if (cycleUnit.equals("Y") && time.length() == 4) {
-            df = new SimpleDateFormat("yyyy");
-        } else if (cycleUnit.equals("M") && time.length() == 6) {
-            df = new SimpleDateFormat("yyyyMM");
-        } else if (cycleUnit.equals("D") && time.length() == 8) {
+        if (cycleUnit.equals(CycleUnitType.DAY) && time.length() == 8) {
```
Comment (aloyszhang) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/DateTransUtils.java:50]:
```suggestion
        if (cycleUnit.equalsIgnoreCase(CycleUnitType.DAY) && time.length() == 8) {
```

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/DateTransUtils.java [hunk 5]
```diff
@@ -45,15 +47,11 @@ public static long timeStrConvertToMillSec(String time, String cycleUnit, TimeZo
...
             df = new SimpleDateFormat("yyyyMMdd");
-        } else if (cycleUnit.equalsIgnoreCase("h") && time.length() == 10) {
+        } else if (cycleUnit.equalsIgnoreCase(CycleUnitType.HOUR) && time.length() == 10) {
             df = new SimpleDateFormat("yyyyMMddHH");
-        } else if (cycleUnit.contains("m") && time.length() == 12) {
+        } else if (cycleUnit.equals(CycleUnitType.MINUTE) && time.length() == 12) {
```
Comment (aloyszhang) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/DateTransUtils.java:54]:
```suggestion
        } else if (cycleUnit.equalsIgnoreCase(CycleUnitType.MINUTE) && time.length() == 12) {
```

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/DateTransUtils.java [hunk 6]
```diff
@@ -77,37 +75,18 @@ public static String millSecConvertToTimeStr(long time, String cycleUnit, TimeZo
...
-        } else if ("M".equals(cycleUnit)) {
-            df = new SimpleDateFormat("yyyyMM");
-        } else if ("D".equalsIgnoreCase(cycleUnit)) {
+        if (CycleUnitType.DAY.equalsIgnoreCase(cycleUnit)) {
             df = new SimpleDateFormat("yyyyMMdd");
-        } else if ("h".equalsIgnoreCase(cycleUnit)) {
+        } else if (CycleUnitType.HOUR.equalsIgnoreCase(cycleUnit)) {
             df = new SimpleDateFormat("yyyyMMddHH");
-        } else if (cycleUnit.contains("m")) {
+        } else if (CycleUnitType.MINUTE.equals(cycleUnit)) {
```
Comment (aloyszhang) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/DateTransUtils.java:82]:
```suggestion
        } else if (CycleUnitType.MINUTE.equalsIgnoreCase(cycleUnit)) {
```

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java (hunk 2 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:200]: m: minute; M: month

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java (hunk 3 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:214]: Due to historical reasons, there were originally situations of 10m and 30m

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/ManagerFetcher.java [hunk 7]
```diff
@@ -227,28 +228,31 @@ private TaskResult getTestConfig(String testDir, int normalTaskId, int retryTask
...
     private DataConfig getTestDataConfig(int taskId, String pattern, boolean retry, Long startTime, Long endTime,
-            int state) {
+            String cycleUnit, int state) {
         DataConfig dataConfig = new DataConfig();
-        dataConfig.setInlongGroupId("testGroupId"); // 老字段 groupId
-        dataConfig.setInlongStreamId("testStreamId"); // 老字段 streamId
-        dataConfig.setDataReportType(1); // 老字段 reportType
+        dataConfig.setInlongGroupId("devcloud_group_id"); // 老字段 groupId
+        dataConfig.setInlongStreamId("devcloud_stream_id"); // 老字段 streamId
+        dataConfig.setDataReportType(0); // 老字段 reportType
```
Comment (aloyszhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/ManagerFetcher.java:241]: Better to explain the field in class `DataConfig` and in english

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/DateTransUtils.java (hunk 5 above)
Comment (justinwwhuang) [reply inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/DateTransUtils.java:54]: The scene of m needs to be case sensitive

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/DateTransUtils.java (hunk 6 above)
Comment (justinwwhuang) [reply inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/DateTransUtils.java:82]: The scene of m needs to be case sensitive

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/task/TestLogFileTask.java [hunk 8]
```diff
@@ -55,68 +58,86 @@ public class TestLogFileTask {
...
-    public void testTaskManager() throws Exception {
-        await().atMost(2, TimeUnit.SECONDS).until(() -> fileName != null && dataTime != null);
-        Assert.assertTrue(fileName.compareTo(resourceName) == 0);
-        Assert.assertTrue(dataTime.compareTo("20230928") == 0);
-        PowerMockito.verifyPrivate(task, Mockito.times(1))
+        await().atMost(10, TimeUnit.SECONDS)
+                .until(() -> fileName.size() == resources.size() && dataTime.size() == resources.size());
+        for (int i = 0; i < fileName.size(); i++) {
+            Assert.assertTrue(fileName.get(i).compareTo(resourceName.get(i)) == 0);
+            Assert.assertTrue(dataTime.get(i).compareTo(srcDataTimes.get(i)) == 0);
```
Comment (aloyszhang) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/task/TestLogFileTask.java:137]:
```suggestion
            Assert.assertEquals(0, fileName.get(i).compareTo(resourceName.get(i)));
            Assert.assertEquals(0, dataTime.get(i).compareTo(srcDataTimes.get(i)));
```

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java (hunk 3 above)
Comment (aloyszhang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:214]: so may be `endwith` is better?

Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java]: ok

**Review by aloyszhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**